xbmc/guilib/GUIIncludes.cpp

```diff
@@ -513,13 +513,22 @@ std::string CGUIIncludes::ResolveConstant(const std::string &constant) const
   return StringUtils::Join(values, ",");
 }
 
-std::string CGUIIncludes::ResolveExpressions(const std::string &expression) const
+std::string CGUIIncludes::ResolveExpressions(const std::string &strInput, const std::vector<std::string> &resolved /* = std::vector<std::string>() */) const
```

Comment:
With it logging `strInput` when it encounters a circular expression, I want it to log the original string that will be easier for skinners to identify, rather than the possibly half-resolved `work`.